db/memtable.cc

```diff
@@ -661,6 +658,14 @@ Status MemTable::Add(SequenceNumber s, ValueType type,
     }
   }
   if (type == kTypeRangeDeletion) {
+    // refresh cached fragmented tombstone list
+    auto* unfragmented_iter =
+        new MemTableIterator(*this, ReadOptions(), nullptr /* arena */,
+                             true /* use_range_del_table */);
+    std::atomic_store(&fragmented_range_tombstone_list_,
```

Comment:
It seems safe to use `std::memory_order_relaxed` here since the new list won't be needed until readers start using seqnos >= this range tombstone, and we already have memory ordering on the seqno. It might not help though since I read atomic shared_ptr is often implemented with global spinlock.

Reply:
This also applies to the std::atomic_load, perhaps more importantly.